### PR TITLE
Update transformers library to 4.57.1 and torch to 2.9.0

### PR DESCRIPTION
**Pull request by jazpurTT**
### Ticket
#1020 

### Problem description
We are updating transformers library to `4.57.1` to keep up to date, broaden model support, and fully support vLLM models. New transformers library performs attention with torch's vmap and torch only added support for exporting/tracing vmap in `torch==2.9.0` so we need to update torch library to 2.9.0

### What's changed
- New transformers library has changed the Cache classes (Static, Dynamic, etc.) to contain and arrays of CacheLayers instead of an array of torch.tensors, we need to extract the torch tensors from the CacheLayers class before comparing values in the comparator.
- Changed reference to key and value cache entries because Cache classes have renamed `kay_cache` -> `keys` and `value_cache` -> `values`. 
- Transformers is now using bitwise logic ops during the causal mask creation on `b8` ops which get converted to `uint8` ops and ttnn does not have support for `uint8` datatype. Added torch decompositions to replace bitwise and/or ops with logical and/or ops when inputs are boolean.
- We started seeing issues when running consecutive tests on llmbox with models with diffrent meshshapes, i.e. (1,8) and (2,4). When ran individually or if we separate them by same mesh size the models are passing. We separated the models into different test groups. We are tracking this issue in #2308 
- Updated torch-xla wheel to a whl built on torch==2.9.0
- Adjusted pcc values for test with small decreases in accuracy.
- non-red llama models, qwen1.5 and phi2 models has a pcc accuracy decrease to a 0.9 - 0.94 range, for now we will `assert_pcc: false` and we tracking the issue in #2543
- Updated test_attention tests to match transformers update.
- we see accuracy drop in llama.py demo, we are tracking issue in #2546
- Changes also need in tt-forge-models repo, we have a PR open that we will uplift into this PR once we get it merged

### Checklist
- [x] New/Existing tests provide coverage for changes

### CODE REVIEW

**Comment by jazpurTT** [comment]
The nightly tests are all passing:
- basic-nightly-test: https://github.com/tenstorrent/tt-xla/actions/runs/20138303727
- model-test-passing: https://github.com/tenstorrent/tt-xla/actions/runs/20100212745
- model-test-full: https://github.com/tenstorrent/tt-xla/actions/runs/20099585568
- model-test-xfail: it is not passing but it matched the last nightly run, same two models failed, [this branch](https://github.com/tenstorrent/tt-xla/actions/runs/20104447480) and [nightly on main](https://github.com/tenstorrent/tt-xla/actions/runs/20117717649)

**Review by jameszianxuTT — APPROVED** [review]
lgtm!

**Review by mmanzoorTT — APPROVED** [review]

File: examples/pytorch/llama.py [hunk 1]
```diff
@@ -171,6 +172,7 @@ def setup_model_and_tokenizer(
     model: torch.nn.Module = AutoModelForCausalLM.from_pretrained(
         model_name, torch_dtype=torch.bfloat16, use_cache=True
     )
+
```
Comment (ajakovljevicTT) [inline examples/pytorch/llama.py]: Leftover from rebasing?

File: tests/infra/testers/single_chip/model/torch_model_tester.py [hunk 2]
```diff
@@ -36,9 +36,9 @@ def masked_is_available():
         try:
             acc = torch.accelerator.current_accelerator()
             # current_accelerator() returns device(type='jax'), need to check .type
-            if acc.type == "jax":
+            if acc is not None and acc.type == "jax":
                 return False
-        except RuntimeError:
+        except (RuntimeError, AttributeError):
```
Comment (ajakovljevicTT) [inline tests/infra/testers/single_chip/model/torch_model_tester.py]: are we getting attribute error even with the if check above?
